src/jsrs_parser.cc

```diff
@@ -375,21 +375,18 @@ Local<Value> ParseString(Isolate*    isolate,
         (string_mode == kApostrophe && begin[i] == '\'')) {
       is_ended = true;
       *size = i + 1;
-      result[res_index] = '\0';
       break;
     }
 
     if (begin[i] == '\\') {
       if (IsLineTerminatorSequence(begin + i + 1, &in_offset)) {
         i += in_offset;
       } else {
-        char* symb =
-            GetControlChar(isolate, begin + ++i, &out_offset, &in_offset);
-        if (!symb) {
+        bool ok = GetControlChar(isolate, begin + ++i, &out_offset, &in_offset,
+                                 result + res_index);
+        if (!ok) {
```

Comment:
Never mind, for some reason it seemed to me that there's one short statement inside the `if` body, so I proposed to eliminate braces.